addons/minedetector/functions/fnc_canDeactivateDetector.sqf

```diff
@@ -1,19 +1,23 @@
 #include "..\script_component.hpp"
 /*
  * Author: Glowbal
- * Check if the mine detector can be deactivated
+ * Checks if the mine detector can be deactivated.
  *
  * Arguments:
- * None
+ * 0: Unit <OBJECT>
+ * 1: Detector type <STRING> (default: currentWeapon Unit)
  *
  * Return Value:
  * Can be deactivated <BOOL>
  *
  * Example:
- * [] call ace_minedetector_fnc_canDeactivateDetector
+ * call ace_minedetector_fnc_canDeactivateDetector
  *
  * Public: No
  */
 
-([ACE_player] call FUNC(hasDetector)) &&
-{[ACE_player, currentWeapon ACE_player] call FUNC(isDetectorEnabled)};
+params ["_unit"];
+
+private _detectorType = param [1, currentWeapon _unit, [""]];
```

Comment:
Why use `param` instead ?
```suggestion
params ["_unit", ["_detectorType", currentWeapon _unit, [""]]];
```

Reply:
Because that doesn't work.

Reply:
What do you mean ?

Reply:
_unit is not yet defined at the point where the array is created